### PR TITLE
fix mailmap parsing issue

### LINKED ISSUE #1416: `gix-mailmap` parser treats mappings with old but not new names as an error
### Current behavior 😯

`<b> test <a>` is silently discarded, unlike `<b> <a>` and `test2 <b> test1 <a>`

### Expected behavior 🤔

Old names, new names, and new emails should be treated uniformly; any combination that includes an old email address should parse and function.

### Git behavior

Git treats this entry as valid and maps signatures accordingly:

```
emily@yuyuko ~/gix-mailmap-bug> git init
Initialized empty Git repository in /Users/emily/gix-mailmap-bug/.git/
emily@yuyuko ~/gix-mailmap-bug (main)> echo '<b> test <a>' >.mailmap
emily@yuyuko ~/gix-mailmap-bug (main)> git check-mailmap 'test <a>'
test <b>
```

### Steps to reproduce 🕹

```
emily@yuyuko ~/gix-mailmap-bug (main)> cat src/main.rs
use gix::actor::SignatureRef;
use gix::mailmap;

fn main() {
    let mailmap = mailmap::Snapshot::from_bytes(b"<b> test <a>");
    let signature = SignatureRef {
        name: "test".into(),
        email: "a".into(),
        time: Default::default(),
    };
    dbg!(mailmap.entries(), signature, mailmap.resolve(signature));
}
emily@yuyuko ~/gix-mailmap-bug (main)> cargo run
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.12s
     Running `target/debug/gix-mailmap-bug`
[src/main.rs:11:5] mailmap.entries() = []
[src/main.rs:11:5] signature = SignatureRef {
    name: "test",
    email: "a",
    time: Time {
        seconds: 0,
        offset: 0,
        sign: Plus,
    },
}
[src/main.rs:11:5] mailmap.resolve(signature) = Signature {
    name: "test",
    email: "a",
    time: Time {
        seconds: 0,
        offset: 0,
        sign: Plus,
    },
}
```

### PR DESCRIPTION
**Pull request by Byron**
Fixes #1416.

### Tasks

* [x] reproduce issue
* [x] fix it
* [x] add `mailmap check` to the CLI

### CODE REVIEW

File: src/plumbing/options/mod.rs [hunk 1]
```diff
@@ -512,10 +512,17 @@ pub mod remote {
 }
 
 pub mod mailmap {
+    use gix::bstr::BString;
+
     #[derive(Debug, clap::Subcommand)]
     pub enum Subcommands {
         /// Print all entries in configured mailmaps, inform about errors as well.
         Entries,
+        /// Print all entries in configured mailmaps, inform about errors as well.
```
Comment (emilazy) [inline src/plumbing/options/mod.rs:521]: The comment here is incorrect.

**Review by emilazy — CHANGES_REQUESTED** [review]
Looks more comprehensive than what I had started to write! I noticed one small issue.

File: src/plumbing/options/mod.rs (hunk 1 above)
Comment (Byron) [reply src/plumbing/options/mod.rs:521]: Right!
Can you make a suggestion? I would just apply it then.

Comment (emilazy) [reply src/plumbing/options/mod.rs:521]: Perhaps:

```suggestion
        /// Print the canonical form of contacts according to the configured mailmaps.
```

**Review by emilazy — CHANGES_REQUESTED** [review]

File: src/plumbing/options/mod.rs (hunk 1 above)
Comment (emilazy) [reply src/plumbing/options/mod.rs:521]: Perhaps:

```suggestion
        /// Print the canonical form of contacts according to the configured mailmaps.
```

**Comment by Byron** [comment]
Thanks for the review!

And here is the new release: https://github.com/Byron/gitoxide/releases/tag/gix-mailmap-v0.23.2

**Pull request merged**